### PR TITLE
fix: return null on null input for timestamp math functions

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 
Checks function inputs for null values

## Testing
QTT and unit tests

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/transform-array.json [hunk 1]
```diff
@@ -93,6 +93,19 @@
...
       ]
+    },
+    {
+      "name": "apply transform lambda function to array of timestamps",
+      "statements": [
+        "CREATE STREAM test (ID STRING KEY, times ARRAY<TIMESTAMP>) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT ID, TRANSFORM(times, x => timestampadd(milliseconds, 5, x)) AS c FROM test;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": "one", "value": {"times": [3, 6]}}
```
Comment (stevenpyzhang) [inline ksqldb-functional-tests/src/test/resources/query-validation-tests/transform-array.json:104]: could we add one more input record where there's a null in the array? I'm curious what the behavior would be, I'd expect the output to just have a null at the same array index but I suspect the entire record value would be null

Comment (jzaralim) [reply ksqldb-functional-tests/src/test/resources/query-validation-tests/transform-array.json]: Added a fix to this PR.

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/timestampsub.json [hunk 2]
```diff
@@ -41,6 +41,23 @@
         {"topic": "TEST2", "key": 0, "value": {"VALUE": 5}},
         {"topic": "TEST2", "key": 0, "value": {"VALUE": -95}}
       ]
+    },
+    {
+      "name": "handles null values",
+      "statements": [
+        "CREATE STREAM TEST (ID INT KEY, time TIMESTAMP, num INTEGER) WITH (kafka_topic='test', value_format='AVRO');",
+        "CREATE STREAM TEST2 AS SELECT id, timestampsub(MILLISECONDS, num, time) AS VALUE FROM TEST;"
```
Comment (stevenpyzhang) [inline ksqldb-functional-tests/src/test/resources/query-validation-tests/timestampsub.json:49]: Not too familiar with the new timestamp code, is milliseconds a reserved keyword in the syntax now for use cases like these?

**Review by stevenpyzhang — APPROVED** [review]
lgtm with build pass

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/timestampsub.json (hunk 2 above)
Comment (jzaralim) [reply ksqldb-functional-tests/src/test/resources/query-validation-tests/timestampsub.json:49]: yup, its the same set of keywords that window expressions use

**Pull request merged**